cloud/docker/docker.py

```diff
@@ -796,13 +804,22 @@ def get_differing_containers(self):
                     continue
 
             # EXPOSED PORTS
-            # Note that ports that are bound at container run are also exposed
-            # implicitly.
-            expected_exposed_ports = set()
+
+            image_exposed_ports = set((image["ContainerConfig"]["ExposedPorts"] or {}).keys())
+
+            expected_exposed_ports = set(image_exposed_ports)
             for p in (self.exposed_ports or []):
                 expected_exposed_ports.add("/".join(p))
 
-            actually_exposed_ports = set((container["Config"]["ExposedPorts"] or {}).keys())
+            if self.port_bindings:
+                for container_port in self.port_bindings.keys():
+                    if isinstance(container_port, int):
+                        container_port = "{}/tcp".format(container_port)
+
+                    expected_exposed_ports.add(container_port)
```

Comment:
Do you mean checking against `container["NetworkSettings"]["Ports"]`? That won't work - I mean comparing:

```python
expected_exposed_ports = set((image['ContainerConfig']['ExposedPorts'] or {}).keys())
for p in (self.exposed_ports or []):
    expected_exposed_ports.add("/".join(p))

actually_exposed_ports = set((container["Config"]["ExposedPorts"] or {}).keys())
```

Image exposed ports + module parameter exposed ports = container exposed ports. Comparing apples to apples, right?

I actually have a PR ready that handles this, environment variables, and volumes from the image configuration. Just a moment, I'll push it up and submit it.